astropy/visualization/wcsaxes/core.py

```diff
@@ -789,11 +787,7 @@ def get_tightbbox(self, renderer, *args, **kwargs):
         bb = [b for b in self._bboxes if b and (b.width != 0 or b.height != 0)]
         bb.append(super().get_tightbbox(renderer, *args, **kwargs))
 
-        if bb:
-            _bbox = Bbox.union(bb)
-            return _bbox
-        else:
-            return self.get_window_extent(renderer)
+        return Bbox.union(bb)
```

Comment:
I think this is fine even though non-trivial because the append above ensures there will always be one element in the list even if it is None for instance, so the list will never be empty